File: internal/shared/model/vendor.go
package model

import "time"

type SupplierVendor struct {
	ID               uint      `gorm:"primaryKey"`
	UserID           uint      `gorm:"uniqueIndex;not null"`
	CompanyName      string    `gorm:"size:128;not null"`
	ContactName      string    `gorm:"size:64"`
	ContactEmail     string    `gorm:"size:128;not null"`
	ContactPhone     string    `gorm:"size:32"`
	BusinessLicense  string    `gorm:"size:256"`
	APIBaseURL       string    `gorm:"size:256;not null"`
	APIAuthType      string    `gorm:"default:api_key;size:32"`
	APIKeyEncrypted  string    `gorm:"size:512"`
	CommissionRate   string    `gorm:"size:16;not null"`
	SettlementCycle  string    `gorm:"default:t+7;size:16"`
	Status           string    `gorm:"not null;default:pending;size:32;index"`
	RejectReason     string    `gorm:"type:text"`
	ApprovedAt       *time.Time
	CreatedAt        time.Time
	UpdatedAt        time.Time
}

type SupplierProduct struct {
	ID             uint      `gorm:"primaryKey"`
	VendorID       uint      `gorm:"index;not null"`
	Name           string    `gorm:"size:128;not null"`
	NameI18n       string    `gorm:"type:text"`
	Type           string    `gorm:"size:64;not null"`
	Model          string    `gorm:"size:64;not null"`
	APIEndpoint    string    `gorm:"size:256;not null"`
	AuthType       string    `gorm:"default:api_key;size:32"`
	Unit           string    `gorm:"size:32;not null"`
	Price          string    `gorm:"size:32;not null"`
	Currency       string    `gorm:"default:USD;size:8"`
	StockTotal     string    `gorm:"size:32"`
	StockRemaining string    `gorm:"size:32"`
	Status         string    `gorm:"default:pending_review;size:32;index"`
	HealthStatus   string    `gorm:"default:unknown;size:32;index"`
	CreatedAt      time.Time
	UpdatedAt      time.Time
}

type Settlement struct {
	ID              uint      `gorm:"primaryKey"`
	VendorID        uint      `gorm:"index;not null"`
	SettlementNo    string    `gorm:"uniqueIndex;size:64;not null"`
	PeriodStart     time.Time `gorm:"not null"`
	PeriodEnd       time.Time `gorm:"not null"`
	TotalSales      string    `gorm:"size:32;not null"`
	CommissionAmount string   `gorm:"size:32;not null"`
	NetAmount       string    `gorm:"size:32;not null"`
	Currency        string    `gorm:"default:USD;size:8"`
	Status          string    `gorm:"default:pending;size:32;index"`
	PaymentMethod   string    `gorm:"size:32"`
	PaidAt          *time.Time
	Remark          string    `gorm:"type:text"`
	CreatedAt       time.Time
	UpdatedAt       time.Time
}

File: web/src/shared/api/vendor.ts
import api from "@/shared/utils/axios";
import type { ApiResponse } from "./types";

export interface VendorProduct {
  id: number;
  vendor_id: number;
  name: string;
  type: string;
  model: string;
  api_endpoint: string;
  unit: string;
  price: string;
  currency: string;
  stock_total?: string;
  stock_remaining?: string;
  status: string;
  health_status: string;
}

export interface VendorSales {
  vendor_id: number;
  period: string;
  total_orders: number;
  total_amount: string;
  paid_orders: number;
  paid_amount: string;
  refunded_orders: number;
  refunded_amount: string;
  by_status: { status: string; count: number; amount: string }[];
}

export interface VendorSettlement {
  id: number;
  vendor_id: number;
  settlement_no: string;
  period_start: string;
  period_end: string;
  total_sales: string;
  commission_amount: string;
  net_amount: string;
  currency: string;
  status: string;
  payment_method?: string;
  paid_at?: string;
  remark?: string;
}

export const vendorService = {
  // Profile
  getProfile: () =>
    api.get<ApiResponse<{ id: number; company_name: string; contact_email: string; status: string }>>("/vendor/me"),

  updateProfile: (data: { contact_name?: string; contact_email?: string; contact_phone?: string; api_base_url?: string }) =>
    api.put<ApiResponse<null>>("/vendor/profile", data),

  apply: (data: {
    company_name: string;
    contact_name: string;
    contact_email: string;
    contact_phone: string;
    business_license: string;
    api_base_url: string;
    api_auth_type: string;
    commission_rate: string;
    settlement_cycle: string;
  }) => api.post<ApiResponse<{ vendor_id: number; status: string }>>("/vendor/apply", data),

  // Products
  listProducts: (params?: { status?: string }) =>
    api.get<ApiResponse<VendorProduct[]>>("/vendor/products", { params }),

  createProduct: (data: {
    name: string;
    type: string;
    model: string;
    api_endpoint: string;
    auth_type?: string;
    unit: string;
    price: string;
    currency?: string;
    stock_total?: string;
  }) => api.post<ApiResponse<VendorProduct>>("/vendor/products", data),

  updateProduct: (id: number, data: { name?: string; price?: string; currency?: string; status?: string }) =>
    api.put<ApiResponse<null>>(`/vendor/products/${id}`, data),

  updateProductPrice: (id: number, price: string) =>
    api.put<ApiResponse<null>>(`/vendor/products/${id}/price`, { price }),

  // Sales
  getSales: (period?: string) =>
    api.get<ApiResponse<VendorSales>>("/vendor/sales", { params: { period } }),

  // Settlements
  getSettlements: () =>
    api.get<ApiResponse<VendorSettlement[]>>("/vendor/settlements"),

  confirmSettlement: (id: number) =>
    api.post<ApiResponse<null>>(`/vendor/settlements/${id}/confirm`),

  requestWithdrawal: (id: number) =>
    api.post<ApiResponse<null>>(`/vendor/settlements/${id}/withdraw`),
};
